src/main/java/org/opensearch/security/filter/SecurityRestFilter.java

```diff
@@ -184,7 +186,9 @@ private boolean checkAndAuthenticateRequest(RestRequest request, RestChannel cha
 
         Matcher matcher = PATTERN_PATH_PREFIX.matcher(request.path());
         final String suffix = matcher.matches() ? matcher.group(2) : null;
-        if(request.method() != Method.OPTIONS && !(HEALTH_SUFFIX.equals(suffix))) {
+        if(request.method() != Method.OPTIONS
+                && !(HEALTH_SUFFIX.equals(suffix))
+                && !(WHO_AM_I_SUFFIX.equals(suffix))) {
```

Comment:
Are these changes needed for replacing TransportClient?

Reply:
They are needed to keep SecurityAdmin to offer:
* whoami https://github.com/opensearch-project/security/blob/04d5dcb6dd6b93b1cfca55c6c41b889da9ccc1a3/src/main/java/org/opensearch/security/tools/SecurityAdmin.java#L244
* healthcheck https://github.com/opensearch-project/security/blob/04d5dcb6dd6b93b1cfca55c6c41b889da9ccc1a3/src/main/java/org/opensearch/security/tools/SecurityAdmin.java#L617